common/lib/plugins/failover2/failover2_plugin.ts

```diff
@@ -0,0 +1,575 @@
+      if (this.isClosed) {
+        await this.invalidInvocationOnClosedConnection();
+      }
+
+      if (this.canUpdateTopology(methodName)) {
+        await this.updateTopology(false);
+      }
+
+      return await methodFunc();
+    } catch (e: any) {
+      logger.debug(Messages.get("Failover.detectedException", e.message));
+      if (this._lastError !== e && this.shouldErrorTriggerClientSwitch(e)) {
+        await this.invalidateCurrentClient();
+        const currentHostInfo = this.pluginService.getCurrentHostInfo();
+        if (currentHostInfo !== null) {
+          this.pluginService.setAvailability(currentHostInfo.allAliases ?? new Set(), HostAvailability.NOT_AVAILABLE);
+        }
+
+        this._lastError = e;
+        await this.pickNewConnection();
+      }
+
+      throw e;
+    }
+  }
+
+  async failover(failedHost: HostInfo) {
+    this.pluginService.setAvailability(failedHost.allAliases, HostAvailability.NOT_AVAILABLE);
+
+    if (this.failoverMode === FailoverMode.STRICT_WRITER) {
+      await this.failoverWriter();
+    } else {
+      await this.failoverReader();
+    }
+
+    if (this._isInTransaction || this.pluginService.isInTransaction()) {
+      // "Transaction resolution unknown. Please re-configure session state if required and try
+      // restarting transaction."
+      logger.debug(Messages.get("Failover.transactionResolutionUnknownError"));
+      throw new TransactionResolutionUnknownError(Messages.get("Failover.transactionResolutionUnknownError"));
+    } else {
+      // "The active SQL connection has changed due to a connection failure. Please re-configure
+      // session state if required."
+      throw new FailoverSuccessError(Messages.get("Failover.connectionChangedError"));
+    }
+  }
+
+  async failoverReader() {
+    const telemetryFactory = this.pluginService.getTelemetryFactory();
+    const telemetryContext = telemetryFactory.openTelemetryContext(Failover2Plugin.TELEMETRY_READER_FAILOVER, TelemetryTraceLevel.NESTED);
+    this.failoverReaderTriggeredCounter.inc();
+
+    const oldAliases = this.pluginService.getCurrentHostInfo()?.allAliases ?? new Set();
+    const failoverEndTimeMs = Date.now() + this.failoverTimeoutSettingMs;
+
+    try {
+      logger.debug(Messages.get("Failover.startReaderFailover"));
+      await telemetryContext.start(async () => {
+        if (!(await this.pluginService.initiateTopologyUpdate(false, this.failoverTimeoutSettingMs))) {
+          // Unable to establish SQL connection to an instance.
+          this.failoverReaderFailedCounter.inc();
+          logger.warn(Messages.get("Failover.unableToConnectToReader"));
+          throw new FailoverFailedError(Messages.get("Failover.unableToConnectToReader"));
+        }
+
+        // Signal to connect that this is an internal call and does not require additional processing.
+        const copyProps = new Map<string, any>(this._properties);
+        copyProps.set(Failover2Plugin.INTERNAL_CONNECT_PROPERTY_NAME, true);
```

Comment:
This new property will cause the community MySQL driver to warn 
```
Ignoring invalid configuration option passed to Connection: 76c06979-49c4-4c86-9600-a63605b83f50. 
```
Please make sure it is handled by `removeWrapperProperties`